### DISCUSSION TITLE
About libseat backend

### DISCUSSION THREAD

**Discussion by fxzxmic** [hyprwm/Hyprland#8421] [Q&A]
```
[LOG] [AQ] [libseat] [libseat/backend/seatd.c:64] Could not connect to socket /run/seatd.sock: No such file or directory
[LOG] [AQ] [libseat] [libseat/libseat.c:76] Backend 'seatd' failed to open seat, skipping
[LOG] [AQ] [libseat] [libseat/libseat.c:76] Backend 'logind' failed to open seat, skipping
[ERR] [AQ] [libseat] [libseat/libseat.c:79] No backend was able to open a seat
[ERR] [AQ] libseat: failed to open a seat
```
If I'm not mistaken, `logind` here refers to `systemd-logind` right? But I am on a systemd system, and `systemd-logind.service` is running, why does logind fail to open the seat?
I know it may not be hyprland related, so if you guys know any info, please let me know, or tell me who I should report the problem to.

**Comment by vaxerski** [comment]
it will only work in a tty, maybe you arent in one?

**Reply by fxzxmic** [reply]
You should be right, I am not in a tty. But seatd is normal in my case. I have reported the problem to systemd and spoken to the seatd developer.

**Reply by vaxerski** [reply]
I think this is an AB problem. What are you trying to do?

**Reply by izmyname** [reply]
Maybe a lil bit of offtopic, but I'm wondering - what does hyprland need seatd for? Doesn't polkit fully replace seatd functionality?

**Reply by vaxerski** [reply]
it doesnt, if you have polkit that will be chosen. We use libinput

**Reply by izmyname** [reply]
So it's safe to wipe seatd dep from both aquamarine and hyprland arch pkgbuilds?

**Reply by vaxerski** [reply]
I think so?

**Reply by fxzxmic** [reply]
Use seatd to open a seat. Systemd can also do it, but since there are some problems with systemd in my environment, I have to use seatd.

**Reply by izmyname** [reply]
Answering my own question:

aquamarine without seatd - builds but doesn't work
Hyprland without seatd, but against aquamarine with seatd - builds, works

**Reply by n-connect** [reply]
@izmyname 

Bumped into similar issue, using Hyperland older version from repo packages, than compiled a whole Hyperland, but before aquamarine, hyperutils, etc. so I can build Hyperland at all. 
Now, olda one runs, new one failed to find DRI/GPU device and starts to complain on seatd first.
```
[LOG] [AQ] [libseat] [libseat/backend/seatd.c:64] Could not connect to socket /run/seatd.sock: No such file or directory
[LOG] [AQ] [libseat] [libseat/libseat.c:76] Backend 'seatd' failed to open seat, skipping
[ERR] [AQ] [libseat] [libseat/backend/logind.c:302] Could not take control of session: Device or resource busy
[LOG] [AQ] [libseat] [libseat/libseat.c:76] Backend 'logind' failed to open seat, skipping
[LOG] [AQ] [libseat] [libseat/libseat.c:73] Seat opened with backend 'builtin'
```
later on
```
[LOG] [AQ] drm: Enumerated device /sys/devices/pci0000:00/0000:00:02.0/drm/card0
[ERR] [AQ] libseat: Couldn't open device at /dev/dri/card0
[ERR] [AQ] drm: Skipping device /sys/devices/pci0000:00/0000:00:02.0/drm/card0, not a KMS device
[ERR] [AQ] drm: Found no gpus to use, cannot continue
[ERR] [AQ] DRM Backend failed
[LOG] [AQ] Starting the Aquamarine backend!
[LOG] [AQ] Starting the Wayland backend!
[LOG] [AQ] Connected to a wayland compositor: GNOME
[LOG] [AQ] Got registry at 0x556eea678330
[LOG] [AQ] Output WAYLAND-1: initialized
[CRITICAL] m_pAqBackend couldn't start! This usually means aquamarine could not find a GPU or encountered some issues. Make s
ure you're running either on a tty or on a Wayland session, NOT an X11 one.
[CRITICAL] Critical error thrown: CBackend::create() failed!
```
Finally the drm and Intel driver already running, but it just works in older Hyprland - so its not because its not there...
```
lsmod | grep drm
drm_buddy              20480  1 i915
drm_display_helper    270336  1 i915
cec                    69632  2 drm_display_helper,i915
drm_kms_helper        241664  2 drm_display_helper,i915
drm                   757760  22 typec_displayport,drm_kms_helper,drm_display_helper,drm_buddy,thinkpad_acpi,i915,ttm
```
Seen similar event at #7423  , but closed a bit prematurely IMHO. Edit: here's another similar issude, but that one solved with a env variable pointed to logind [Error starting Hyprland: something wrong with libseat. [SOLVED]](https://bbs.archlinux.org/viewtopic.php?id=290280)
Anyway, reading your last answer is the best source up to now. The interesting part, I guess I've never had seatd installed before just logind, but also my old Hyperland do not use aquamarine yet. Do you have any idea how-to compile aquamarine, so it can work with logind too/let use the working GPU/DRM device?

**Comment by n-connect** [comment]
After some back & forth, did the following in another session instead of Hyperland 
- put all newly compiled Hyperland v0.47.1 under /usr/local/*  (bin, include & share dirs' contents)
- renamed old 3 binaries and edit the original softlink called hyprland points to new
- installed seatd and started the service -> 
 
Started Hyprland manually first from GNOME in a terminal, then with login from gdm
Finally it worked, just some windowrules needed to be updated for v2 version.

Key points:
- do not test new Hyprland from Hyperland session :)
- copy over / extract everything in once, not just the binaries

Short of the above will give false error messages. I do not ever sure seatd needed, will check later
